### PR TITLE
Remove Maker discount eligibility application and discount code

### PR DESCRIPTION
**Pull request by TurnerRiley**
The Maker discount eligibility application and discount code (previously found through [studio.code.org/maker/home](https://studio.code.org/maker/home)) have been deprecated and this PR removes the remaining code referencing the application and code. It also seems like the process for acquiring the eligibility code involved obtaining and displaying a login code (so that's been removed in this PR as well).

Note: "maker" is also referred to as "circuit playground" and "unit 6" in some places in case you see deletions involving those names.

## Links
Jira ticket: [here](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?assignee=60d62161f65054006980bd71&selectedIssue=ACQ-347)

## Testing story
Passing drone build.

## Follow-up work
The RED team said they no longer have use for the `CircuitPlaygroundDiscountApplication` and `CircuitPlaygroundDiscountCode` tables, so I will create a followup PR that creates a migration to drop those tables.

DCDO flags to delete once we have support for removing them:
- facilitator_ids_eligible_for_maker_discount
- currently_distributing_discount_codes

### CODE REVIEW

File: dashboard/app/controllers/maker_controller.rb [hunk 1]
```diff
@@ -1,6 +1,4 @@
 class MakerController < ApplicationController
-  authorize_resource class: :maker_discount, except: [:home, :setup, :login_code, :display_code, :confirm_login]
```
Comment (bethanyaconnor) [inline dashboard/app/controllers/maker_controller.rb:2]: Can you explain a bit why we can remove routes like `login_code` when they're not using the MakerDiscount class? I don't know what they do and the comments aren't super clear to me.

File: dashboard/config/routes.rb [hunk 2]
```diff
@@ -200,7 +191,6 @@
       get '/reset_session', to: 'sessions#reset'
       get '/lockout', to: 'sessions#lockout'
       get '/users/existing_account', to: 'registrations#existing_account'
-      post '/users/auth/maker_google_oauth2', to: 'omniauth_callbacks#maker_google_oauth2'
```
Comment (bethanyaconnor) [inline dashboard/config/routes.rb:203]: are we sure this was only used for the discount?

File: lib/cdo/delete_accounts_helper.rb [hunk 3]
```diff
@@ -287,16 +287,6 @@ def remove_email_preferences(email)
     @log.puts "Removed #{record_count} EmailPreference" if record_count > 0
   end
 
-  # Removes signature and school_id from applications for this user
-  # @param [User] user
-  def anonymize_circuit_playground_discount_application(user)
```
Comment (bethanyaconnor) [inline lib/cdo/delete_accounts_helper.rb:292]: This can be deleted because we're about to completely drop the tables right?

**Review by bethanyaconnor — APPROVED** [review]
Overall looks good and I love a huge negative diff 🤩 I wonder if it's worth tagging student learning on this PR as well as a gut check that none of the auth routes are used anywhere?

**Comment by bencodeorg** [comment]
I've been in this code before and would be happy to take a look :). It looks like things got funky with the rebase though?

**Comment by TurnerRiley** [comment]
> I've been in this code before and would be happy to take a look :). It looks like things got funky with the rebase though?

Oh whoops yes. Once we get staging-next into staging, I'll clean this back up and re-request :) Thank you!

**Comment by TurnerRiley** [comment]
> > I've been in this code before and would be happy to take a look :). It looks like things got funky with the rebase though?
> 
> Oh whoops yes. Once we get staging-next into staging, I'll clean this back up and re-request :) Thank you!

Closing this PR and sort of re-creating it for a few reasons: first, I was having trouble with getting the PR lined up with staging after rebasing from staging-next (and I was spending more time fixing this than I figured it would take to re-set it up), and I had reason to reset it since I think there are too many nearby pieces I tried to pull into this deletion when I think it would be easier to review (and potentially revert should anything go wrong) if I split it up into pieces.

**Pull request closed without merging**